spec/datadog/opentelemetry/metrics_spec.rb

```diff
@@ -0,0 +1,344 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+# OpenTelemetry metrics SDK requires Ruby >= 3.1
+if Gem::Version.new(RUBY_VERSION) >= Gem::Version.new('3.1')
+  require 'opentelemetry/sdk'
+  require 'opentelemetry-metrics-sdk'
+  require 'opentelemetry/exporter/otlp_metrics'
+end
+
+require 'datadog/opentelemetry'
+require 'datadog/core/configuration/settings'
+require 'net/http'
+require 'json'
+
+DEFAULT_OTLP_HTTP_PORT = 4318
+
+RSpec.describe 'OpenTelemetry Metrics Integration', ruby: '>= 3.1' do
+  include NetworkHelpers
+
+  before do
+    Datadog.send(:reset!) if Datadog.respond_to?(:reset!, true)
+    provider = ::OpenTelemetry.meter_provider
+    provider.shutdown if provider.is_a?(::OpenTelemetry::SDK::Metrics::MeterProvider)
+    ::OpenTelemetry.meter_provider = ::OpenTelemetry::Internal::ProxyMeterProvider.new
+    allow(Datadog.logger).to receive(:warn)
+    allow(Datadog.logger).to receive(:error)
+    allow(Datadog.logger).to receive(:debug)
+    clear_testagent_metrics
+  end
+
+  after do
+    provider = ::OpenTelemetry.meter_provider
+    # Ensures background threads collecting metrics are shutdown.
+    provider.shutdown if provider.is_a?(::OpenTelemetry::SDK::Metrics::MeterProvider)
+    clear_testagent_metrics
+  end
+
+  def clear_testagent_metrics
+    uri = URI("http://#{agent_host}:#{DEFAULT_OTLP_HTTP_PORT}/test/session/clear")
+    Net::HTTP.post_form(uri, {})
+  rescue => e
+    puts "Error clearing testagent metrics: #{e.message}"
+  end
```

Comment:
If there's errors cleaning this up, should we just fail the test?

Reply:
Previously I was attempting to clear the metrics stored in the testagent before and after a test run and hoping at lest one of these requests would be successful. Now that we only do the reset in the before block we should fail if the testagent is not available or can't handle the request.